### PR TITLE
brainflow: init at 5.11.0

### PR DESCRIPTION
**Pull request by Jingles52**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/brainflow/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "brainflow";
+  version = "5.10.2";
+
+  src = fetchFromGitHub {
+    owner = "brainflow-dev";
+    repo = "brainflow";
+    rev = "${version}";
+    sha256 = "sha256-8SZDma05imXcCW/NbRjQTkxL9gKSDCQ64yG6E+GhevA=";
```
Comment (dotlambda) [inline pkgs/development/libraries/brainflow/default.nix:11]:
```suggestion
    hash = "sha256-8SZDma05imXcCW/NbRjQTkxL9gKSDCQ64yG6E+GhevA=";
```

File: pkgs/development/libraries/brainflow/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  meta = with lib; {
+    description = "BrainFlow is a library intended to obtain, parse and analyze EEG, EMG, ECG and other kinds of data from biosensors.";
```
Comment (dotlambda) [inline pkgs/development/libraries/brainflow/default.nix:17]:
```suggestion
    description = "Obtain, parse and analyze EEG, EMG, ECG and other kinds of data from biosensors";
```

File: pkgs/development/libraries/brainflow/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    homepage = "https://brainflow.org/";
+    license = licenses.mit;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ ziguana ];
```
Comment (dotlambda) [inline pkgs/development/libraries/brainflow/default.nix:21]: We usually order the attributes alphabetically.

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -18452,6 +18452,8 @@ with pkgs;
 
   aws-adfs = with python3Packages; toPythonApplication aws-adfs;
 
+  brainflow = callPackage ../development/libraries/brainflow { };
```
Comment (dotlambda) [inline pkgs/top-level/all-packages.nix:18455]: Use `pkgs/by-name` instead.

File: pkgs/development/libraries/brainflow/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{ lib, stdenv, fetchFromGitHub, cmake }:
+
+stdenv.mkDerivation rec {
+  pname = "brainflow";
+  version = "5.10.2";
+
+  src = fetchFromGitHub {
+    owner = "brainflow-dev";
+    repo = "brainflow";
+    rev = "${version}";
```
Comment (dotlambda) [inline pkgs/development/libraries/brainflow/default.nix:10]:
```suggestion
    rev = version;
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (ziguana) [reply pkgs/top-level/all-packages.nix:18455]: Done.

File: pkgs/development/libraries/brainflow/default.nix (hunk 3 above)
Comment (ziguana) [reply pkgs/development/libraries/brainflow/default.nix]: Moved `platforms` down, I think that's what you meant.

**Review by dotlambda — APPROVED** [review]
LGTM though it would be nice if `$out/inc` were `$out/include`.

**Comment by ziguana** [comment]
bumped the version and patched the include destination

File: pkgs/by-name/br/brainflow/package.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  postPatch = ''
+    for file in \
+      "cpp_package/build.cmake" \
+      "src/ml/build.cmake" \
+      "src/board_controller/openbci/ganglion_bglib/build.cmake" \
+      "src/utils/bluetooth/build.cmake" \
+      "src/board_controller/build.cmake" \
+      "src/board_controller/neuromd/brainbit_bglib/build.cmake" \
+      "src/board_controller/muse/muse_bglib/build.cmake" \
+      "src/data_handler/build.cmake"; do
```
Comment (dotlambda) [inline pkgs/by-name/br/brainflow/package.nix:25]: Does
```suggestion
    for file in **/build.cmake; do
```
work?

Comment (ziguana) [reply pkgs/by-name/br/brainflow/package.nix:25]: not completely, it produces the output below

```
/nix/store/ainnpdma6klsb6wky4phj1n1z2zf0bqa-brainflow-5.11.0
 ├──inc
 │  ├──board_controller.h
 │  ├──board_info_getter.h
 │  ├──brainflow_array.h
 │  ├──brainflow_constants.h
 │  ├──brainflow_exception.h
 │  ├──brainflow_input_params.h
 │  ├──brainflow_model_params.h
 │  ├──data_handler.h
 │  ├──json.hpp
 │  ├──ml_module.h
 │  └──shared_export.h
 ├──include
 │  ├──board_shim.h
 │  ├──brainflow_array.h
 │  ├──brainflow_exception.h
 │  ├──data_filter.h
 │  └──ml_model.h
...
```

Comment (ziguana) [reply pkgs/by-name/br/brainflow/package.nix:25]: `find`/`sed` works, do you want to go with that?

```
    find . -type f -name 'build.cmake' -exec \
      sed -i 's/DESTINATION inc/DESTINATION include/g' {} \;
```

File: pkgs/by-name/br/brainflow/package.nix [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+          --replace "DESTINATION inc" "DESTINATION include"
+    done
+  '';
+
+  meta = with lib; {
+    description = "Obtain, parse and analyze EEG, EMG, ECG and other kinds of data from biosensors";
+    homepage = "https://brainflow.org/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ziguana ];
+    platforms = platforms.linux;
```
Comment (AndersonTorres) [inline pkgs/by-name/br/brainflow/package.nix:36]: mainProgram?

File: pkgs/by-name/br/brainflow/package.nix (hunk 6 above)
Comment (AndersonTorres) [reply pkgs/by-name/br/brainflow/package.nix:25]: I like the more explicit approach. Keep it.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/br/brainflow/package.nix [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+      "cpp_package/build.cmake" \
+      "src/ml/build.cmake" \
+      "src/board_controller/openbci/ganglion_bglib/build.cmake" \
+      "src/utils/bluetooth/build.cmake" \
+      "src/board_controller/build.cmake" \
+      "src/board_controller/neuromd/brainbit_bglib/build.cmake" \
+      "src/board_controller/muse/muse_bglib/build.cmake" \
+      "src/data_handler/build.cmake"; do
+        substituteInPlace "$file" \
+          --replace "DESTINATION inc" "DESTINATION include"
```
Comment (dotlambda) [inline pkgs/by-name/br/brainflow/package.nix:27]:
```suggestion
          --replace-fail "DESTINATION inc" "DESTINATION include"
```

File: pkgs/by-name/br/brainflow/package.nix (hunk 7 above)
Comment (ziguana) [reply pkgs/by-name/br/brainflow/package.nix:36]: it's a library

**Pull request closed without merging**